### DISCUSSION TITLE
Introduce ProcessManager mixin similar to AggregateRoot in order to help with event sourced process managers

### DISCUSSION THREAD

**Discussion by mostlyobvious** [RailsEventStore/rails_event_store#1886] [Ideas]
Possible implementation in: https://blog.arkency.com/process-managers-revisited/

**Comment by mostlyobvious** [comment]
Possibly related to #335

**Comment by tomaszwro** [comment]
after discussing with @pawelpacana: let's make something like https://github.com/arkency/aggregates, but for process managers

good example of a process manager is in https://github.com/RailsEventStore/cqrs-es-sample-with-res
